en/Basic Math/magic_square.md

```diff
@@ -0,0 +1,62 @@
+# What is a Magic Square Puzzle?
+A magic square is defined as a square containing several distinct integers arranged so that the total or sum of the numbers is the same in every row, column, and main diagonal and usually in some or all of the other diagonals.
+
+# Magic Square Formula
+A magic square puzzle of the order n is an organization of n2 numbers, usually unique integers, in a square. The n numbers in all rows, columns, and diagonals add up to the same constant. A magic square bears the integers from 1 to n2. The fixed sum in every row, column and diagonal is known as the magic constant or the magic sum. It is represented by the letter M. The magic constant of a typical magic square depends entirely on the value of n. Thus, the value of the magic sum is calculated using the formula:
+- M = n(n2 + 1)/2
+- This is the formula for a magic square that is used to make magic squares of different orders. If we subtract each number from (n2 + 1), we get another magic square, and this is called the complementary magic square. A square containing consecutive numbers beginning with 1 is often called the normal magic square.
+
+# How to Solve Magic square
+As mentioned above, the formula of the magic square sum is n(n2 + 1)/2.
+
+For a magic square of order 3, we need to substitute n = 3 to know the magic sum so that we can easily form the magic square 3×3.
+
+When n = 3, the sum = 3(32 + 1) = 3(9 + 1)/2 = (3 × 10)/2 = 15
+
+Now, we have to place the numbers in the respective places so that the sum of numbers in each row, column and diagonal is equal to 15.
```

Comment:
```suggestion
# How to Solve Magic square

As mentioned above, the formula of the magic square sum is n(n2 + 1)/2.\
For a magic square of order 3, we need to substitute `n = 3` to know the magic sum so that we can easily form the magic square 3×3.

When `n = 3`, the sum = 3(32 + 1) = 3(9 + 1)/2 = (3 × 10)/2 = 15\
Now, we have to place the numbers in the respective places so that the sum of numbers in each row, column, and diagonal is equal to 15.
```